pkg/reader/ccip.go

```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
 	return addressBytes, nil
 }
 
+// GetLinkPriceUSD gets the LINK price in USD from the FeeQuoter contract on the destination chain.
+func (r *ccipChainReader) GetLinkPriceUSD(ctx context.Context) (cciptypes.BigInt, error) {
+	// Exit without an error if we cannot read the destination.
+	if err := validateExtendedReaderExistence(r.contractReaders, r.destChain); err != nil {
+		return cciptypes.BigInt{}, fmt.Errorf("validate extended reader existence: %w", err)
+	}
+
+	feeQuoterCfg, err := r.getDestFeeQuoterStaticConfig(ctx)
+	if err != nil {
+		return cciptypes.BigInt{}, fmt.Errorf("get destination fee quoter static config: %w", err)
+	}
+
+	linkTokenAddr := feeQuoterCfg.LinkToken
+
+	getLinkPriceUSD, err := r.getLinkPriceUSD(ctx, linkTokenAddr)
+
+	if err != nil {
+		return cciptypes.BigInt{}, fmt.Errorf("get LINK price in USD: %w", err)
+	}
+
+	return getLinkPriceUSD, nil
+}
+
+// getDestFeeQuoterStaticConfig returns the destination chain's Fee Quoter's StaticConfig
+func (r *ccipChainReader) getDestFeeQuoterStaticConfig(ctx context.Context) (feeQuoterStaticConfig, error) {
+	var staticConfig feeQuoterStaticConfig
+	err := r.getDestinationData(
+		ctx,
+		r.destChain,
+		consts.ContractNameFeeQuoter,
+		consts.MethodNameFeeQuoterGetStaticConfig,
+		&staticConfig,
+	)
+
+	if err != nil {
+		return feeQuoterStaticConfig{}, fmt.Errorf("unable to lookup nonce manager (offramp static config): %w", err)
+	}
+
+	return staticConfig, nil
+}
+
+// TODO: doc
+func (r *ccipChainReader) getLinkPriceUSD(ctx context.Context, linkTokenAddr []byte) (cciptypes.BigInt, error) {
+	if len(linkTokenAddr) == 0 {
+		return cciptypes.BigInt{}, fmt.Errorf("linkTokenAddr is empty")
+	}
+
+	reader, ok := r.contractReaders[r.destChain]
+	if !ok {
+		return cciptypes.BigInt{}, fmt.Errorf("contract reader not found for chain %d", r.destChain)
+	}
```

Comment:
This is the second time this check is done in this call stack, dedup?

Reply:
The alternative is to get rid of the `validateExtendedReaderExistence` call, but that's a stronger check. I'm fine with leaving both in.

Reply:
This function is a duplicate from `OnchainTokenPricesReader.GetFeeQuoterTokenUpdates`. Maybe pass the price reader to CCIP and use it instead.

Reply:
`GetWrappedNativeTokenPriceUSD` also makes the same FeeQuoter call, and there's other FeeQuoter calls in `ccipChainReader`. I think it makes sense to consolidate FeeQuoter functionality in a future PR, "after the dust settles".